src/nbl/ui/CWindowWin32.cpp

```diff
@@ -106,9 +109,37 @@ namespace ui
 				windowLeft = 0;
 				windowTop = 0;
 			}
+			{
+				//TODO: thoroughly test this stuff	
+				constexpr uint32_t INPUT_DEVICES_COUNT = 5;
+				RAWINPUTDEVICE inputDevices[INPUT_DEVICES_COUNT];
+				inputDevices[0].hwndTarget = m_params.nativeWindow;
+				inputDevices[0].dwFlags = RIDEV_DEVNOTIFY;
+				inputDevices[0].usUsagePage = HID_USAGE_PAGE_GENERIC;
+				inputDevices[0].usUsage = HID_USAGE_GENERIC_POINTER;
+
+				inputDevices[1].hwndTarget = m_params.nativeWindow;
+				inputDevices[1].dwFlags = RIDEV_DEVNOTIFY;
+				inputDevices[1].usUsagePage = HID_USAGE_PAGE_GENERIC;
+				inputDevices[1].usUsage = HID_USAGE_GENERIC_MOUSE;
+
+				inputDevices[2].hwndTarget = m_params.nativeWindow;
+				inputDevices[2].dwFlags = RIDEV_DEVNOTIFY;
+				inputDevices[2].usUsagePage = HID_USAGE_PAGE_GENERIC;
+				inputDevices[2].usUsage = HID_USAGE_GENERIC_KEYBOARD;
 
-			// create window
+				inputDevices[3].hwndTarget = m_params.nativeWindow;
+				inputDevices[3].dwFlags = RIDEV_DEVNOTIFY;
+				inputDevices[3].usUsagePage = HID_USAGE_PAGE_GAME;
+				inputDevices[3].usUsage = HID_USAGE_GENERIC_JOYSTICK;
 
+				inputDevices[4].hwndTarget = m_params.nativeWindow;
+				inputDevices[4].dwFlags = RIDEV_DEVNOTIFY;
+				inputDevices[4].usUsagePage = HID_USAGE_PAGE_GAME;
+				inputDevices[4].usUsage = HID_USAGE_GENERIC_GAMEPAD;
+
+				RegisterRawInputDevices(inputDevices, INPUT_DEVICES_COUNT, sizeof(RAWINPUTDEVICE));
```

Comment:
what does it do?

Reply:
> what does it do?

Registering devices i want to receive connect/disconnect events from.